qt/src/SourceManager.cc

```diff
@@ -44,6 +54,32 @@ QDataStream& operator>>(QDataStream& ds, Source*&) {
 	return ds;
 }
 
+Source::Source(const QString& _path, const QString& _displayname, bool _isTemp)
+				: path(_path), displayname(_displayname), isTemp(_isTemp) {
+	// Check whether input PDF file is encrypted
+	if(path.endsWith(".pdf", Qt::CaseInsensitive)) {
+		std::unique_ptr<Poppler::Document> document(Poppler::Document::load(path));
+		if(document && document->isLocked()) {
+			bool ok = false;
+			QString text;
+			while(true) {
+				text = QInputDialog::getText(MAIN, _("Enter password for the PDF file"),
+											 _("Password:"), QLineEdit::Password,
+											 "", &ok);
+				if(!ok) {
+					throw std::invalid_argument("Locked PDF: skip file.");
+				}
+				if(!document->unlock(text.toLocal8Bit(), text.toLocal8Bit())) {
+					password = text.toLocal8Bit();
+					break;
+				} else {
+					QMessageBox::information(MAIN, _("Open encrypted PDF"), _("Entered password is wrong."));
+				}
```

Comment:
The double dialog is kinda annoying.. What do you think about just showing the password dialog again, so that users who mistyped can just continue typing the next attempt instead of having to click away the info dialog? Many other applications also just keep showing the password dialog until the input is correct or the user explicitly cancels.

Reply:
Oh... i thought about it. But with additional MessageBox i give to user some feedback.

Reply:
You could pass the previously entered to the `QInputDialog::getText`, so that the user will see that the dialog is re-shown with the previous input selected, which is pretty standard behaviour and should make users realize that the input was wrong.

Reply:
Ok, i disabled additional messagebox